sdk/python/kfp/compiler/compiler_test.py

```diff
@@ -3080,5 +3080,23 @@ def my_pipeline(sample_input1: str = 'message'):
             .strategy, 0)
 
 
+class TestOutputDefinitionsPresentWhenCompilingComponents(unittest.TestCase):
+
+    def test_basic_permitted(self):
+
+        @dsl.component
+        def comp(message: str) -> str:
+            return message
+
+        with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
It's possible to assert on `identity.pipeline_spec.root.output_definitions` directly, which avoids the compile/load steps